### PR TITLE
[Improvement][Task Plugin] Support HTTP Task Plugin output response as a parameter 

### LINKED ISSUE #13528: [Improvement][Task Plugin] Support Task Plugin output response as a parameter
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

image

The downstream task **shell**
image

The downstream task **python**
image

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jieguangzhou**
## Purpose of the pull request

close: #13528

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: docs/docs/en/guide/task/http.md [hunk 1]
```diff
@@ -16,12 +16,24 @@ This node is used to perform http type tasks such as the common POST and GET req
...
-| Verification content | When the verification condition selects a custom response code, the content contains, and the content does not contain, the verification content is required. |
-| Custom parameter     | It is a user-defined parameter of http part, which will replace the content with `${variable}` in the script.                                                 |
+|      **Parameter**      |                                                                        **Description**                                                                        |
+|-------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Request address         | HTTP request URL.                                                                                                                                             |
+| Request type            | Supports GET, POSt, HEAD, PUT, DELETE.                                                                                                                        |
```
Comment (SbloodyS) [inline docs/docs/en/guide/task/http.md]:
```suggestion
| Request type            | Supports GET, POST, HEAD, PUT, DELETE.                                                                                                                        |
```

File: docs/docs/zh/guide/task/http.md [hunk 2]
```diff
@@ -26,6 +26,16 @@
 | 校验内容     | 当校验条件选择自定义响应码、内容包含、内容不包含时，需填写校验内容   |
 | 自定义参数    | 是 http 局部的用户自定义参数，会替换脚本中以 ${变量} 的内容 |
 
+## 任务输出参数
+
+| **任务参数** |       **描述**        |
+|----------|---------------------|
+| response | VARCHAR, http请求返回结果 |
```
Comment (SbloodyS) [inline docs/docs/zh/guide/task/http.md:33]:
```suggestion
| **任务参数** |       **描述**        |
|--------------|-----------------|
| response | VARCHAR, http请求返回结果 |
```

**Review by SbloodyS — COMMENTED** [review]
Generally LGTM. Just some NIT.

File: docs/docs/en/guide/task/http.md (hunk 1 above)
Comment (jieguangzhou) [reply docs/docs/en/guide/task/http.md]: Done

**Review by SbloodyS — DISMISSED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/test/java/org/apache/dolphinscheduler/plugin/task/sqoop/EntityTestUtils.java [hunk 3]
```diff
@@ -75,6 +77,9 @@ public static void run(List<Class> classList) throws IllegalAccessException, Inv
                     for (int i = 0; i < parameterTypes.length; i++) {
                         objects[i] = OBJECT_MAP.get(parameterTypes[i].getName());
                     }
+                    if (ObjectUtils.allNull(objects)) {
```
Comment (zhongjiajie) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/test/java/org/apache/dolphinscheduler/plugin/task/sqoop/EntityTestUtils.java:80]: why we add this code?

Comment (zhongjiajie) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sqoop/src/test/java/org/apache/dolphinscheduler/plugin/task/sqoop/EntityTestUtils.java:80]: ok, then LGTM now

**Comment by jieguangzhou** [comment]
@zhongjiajie @SbloodyS Can we mark this pr as a bugfix, and release it in version 3.1.4

**Review by EricGao888 — APPROVED** [review]
LGTM, but why do u think it is a bugfix? It looks more like a feature or improvement to me.

**Comment by jieguangzhou** [comment]
> LGTM, but why do u think it is a bugfix? It looks more like a feature or improvement to me.

Yes, It is an improvement. I'm not sure if there will be any minor improvements in version z

**Comment by jieguangzhou** [comment]
BTW, Do we need to provide custom output in more task plugins? Such as some remote tasks, we can output the appId to the downstream tasks.

For example, we can output the pod names in the k8s task

**Pull request merged**